File: rb/features.rb
# Scryfall SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ScryfallFeatures
  def self.make_feature(name)
    case name
    when "base"
      ScryfallBaseFeature.new
    when "ratelimit"
      ScryfallRatelimitFeature.new
    when "retry"
      ScryfallRetryFeature.new
    when "test"
      ScryfallTestFeature.new
    when "timeout"
      ScryfallTimeoutFeature.new
    else
      ScryfallBaseFeature.new
    end
  end
end

File: py/features.py
# Scryfall SDK feature factory

from feature.base_feature import ScryfallBaseFeature
from feature.test_feature import ScryfallTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ScryfallBaseFeature(),
        "test": lambda: ScryfallTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
